buildSrc/src/main/kotlin/com/datadog/gradle/config/JacocoConfig.kt

```diff
@@ -9,77 +9,58 @@ package com.datadog.gradle.config
 import org.gradle.testing.jacoco.tasks.JacocoCoverageVerification
 import org.gradle.testing.jacoco.tasks.JacocoReport
 
 fun Project.jacocoConfig() {
 
-    val jacocoTestDebugUnitTestReport = tasks.create("jacocoTestDebugUnitTestReport", JacocoReport::class.java)
-    jacocoTestDebugUnitTestReport.reports {
+    val jacocoTestReport = tasks.getByName("jacocoTestReport", JacocoReport::class)
+    jacocoTestReport.reports {
         csv.isEnabled = false
         xml.isEnabled = true
         html.isEnabled = true
-        html.destination = file("${buildDir.path}/reports/jacoco/jacocoTestDebugUnitTestReport/html")
+        html.destination = file("${buildDir.path}/reports/jacoco/jacocoTestReport/html")
     }
 
-    val jacocoTestReleaseUnitTestReport = tasks.create("jacocoTestReleaseUnitTestReport", JacocoReport::class.java)
-    jacocoTestReleaseUnitTestReport.reports {
-        csv.isEnabled = false
-        xml.isEnabled = true
-        html.isEnabled = true
-        html.destination = file("${buildDir.path}/reports/jacoco/jacocoTestReleaseUnitTestReport/html")
-    }
-
-    val jacocoTestCoverageVerification =
-        tasks.create("jacocoTestCoverageVerification", JacocoCoverageVerification::class.java)
+    val jacocoTestCoverageVerification = tasks.getByName("jacocoTestCoverageVerification", JacocoCoverageVerification::class)
     jacocoTestCoverageVerification.violationRules {
         rule {
             limit {
-                minimum = BigDecimal(0.85)
+                // TODO increase that when coverage is better?
+                minimum = BigDecimal(0.75)
             }
         }
     }
 
     listOf(
-        jacocoTestDebugUnitTestReport,
-        jacocoTestReleaseUnitTestReport,
+        jacocoTestReport,
         jacocoTestCoverageVerification
     ).forEach { task ->
-        val excludeFilters = arrayOf(
-            "**/R.class",
-            "**/R$*.class",
-            "**/BuildConfig.*",
-            "**/Manifest*.*",
-            "**/*Test*.*",
-            "android/**/*.*",
-            "**/data/models/*"
-        )
 
-        val debugTree = fileTree("${buildDir.path}/intermediates/classes/debug").apply {
-            exclude(*excludeFilters)
-        }
-        val kotlinDebugTree = fileTree("${buildDir.path}/tmp/kotlin-classes/debug").apply {
-            exclude(*excludeFilters)
-        }
         val mainSrc = "${project.projectDir}/src/main/kotlin"
 
-        task.classDirectories.setFrom(files(debugTree, kotlinDebugTree))
-        task.executionData.setFrom(files("${buildDir.path}/jacoco/testDebugUnitTest.exec"))
+        task.executionData.setFrom(files("${buildDir.path}/jacoco/test.exec"))
         task.sourceDirectories.setFrom(files(mainSrc))
     }
-    jacocoTestDebugUnitTestReport.dependsOn("testDebugUnitTest")
-    jacocoTestReleaseUnitTestReport.dependsOn("testReleaseUnitTest")
-    jacocoTestCoverageVerification.dependsOn(jacocoTestDebugUnitTestReport)
+    jacocoTestReport.dependsOn("test")
```

Comment:
here you are making the `jacocoTestReport` to depend on `test` but bellow you actually finalize the `test` task with `jacocoTestReport` . Isn't this going to create a circular dependency ?

Reply:
no, it is fine. `dependsOn` just means that this task should be executed by the time parent task is triggered.

Gradle's DAG is resolved fine because in gradle task name is unique, so if we run `jacocoTestReport` it will run `test` and by the completion of `test` gradle knows that `jacocoTestReport` is in the pending state and its dependencies just completed and should not be run again.